### ISSUE TITLE
memory mapped file not deleted with forced process termination

### ISSUE THREAD

**Issue by jepickett** [microsoftarchive/redis#93]
The QFork memory mapped file is created with FILE_FLAG_DELETE_ON_CLOSE. Normally this erases the file when the process exits. In the case of forced process termination, TerminateProcess() or a power failure, the memory mapped file is not deleted. Since the file can be very large, there should be some mechanism for detecting the orphaned file and deleting it when Redis starts up again.

**Comment by JCMais** [comment]
One of my hard drives had been constantly running out of space, and the reason was just that, the memory mapped files were not being removed by Redis. I've created a script just to clean old files after redis starts:

``` Batchfile
@ECHO off
SETLOCAL enabledelayedexpansion

FOR %%f in (*.dat) do (

    SET REDISFILE=%%~nf

    FOR /F "tokens=1* delims=_" %%a in ( "!REDISFILE!" ) do (

        tasklist /FI "PID eq %%b" 2>NUL | find /I /N "%%b">NUL

        IF NOT !ERRORLEVEL! == 0 (
            rm %%f
        )
    )

)
```

It basically gets the PID from each file, check if the PID is active, if not, remove the file.

**Comment by jepickett** [comment]
The behavior described in this issue would require a call to TerminateProcess(), or a physical power removal in order to manifest. Why is Redis getting killed frequently enough in your environment to fill the disk?

**Comment by JCMais** [comment]
In reality nothing that can (should) happen on production servers. Power outages and BSOD.

**Comment by jepickett** [comment]
Fixed in commit 4f0e17fb35256877f8a7677a098b481dfb9b6b66
